validator/api/src/main/java/tech/pegasys/artemis/validator/api/ValidatorDuties.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.api;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.primitives.UnsignedLong;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import tech.pegasys.artemis.util.bls.BLSPublicKey;
+
+public class ValidatorDuties {
+  private final BLSPublicKey publicKey;
+  private final Optional<Duties> duties;
+
+  private ValidatorDuties(final BLSPublicKey publicKey, final Optional<Duties> duties) {
+    this.publicKey = publicKey;
+    this.duties = duties;
+  }
+
+  public static ValidatorDuties forKnownValidator(
```

Comment:
nit: 
```suggestion
  public static ValidatorDuties forValidatorWithKnownDuties(
```

Reply:
Wound up changing both so we have `noDuties` and `withDuties`.